raster/src/main/scala/geotrellis/raster/io/geotiff/ByteGeoTiffSegment.scala

```diff
@@ -4,57 +4,24 @@ import geotrellis.raster._
 import geotrellis.raster.io.geotiff.utils._
 import geotrellis.raster.io.geotiff.compression._
 
-import java.util.BitSet
-
 import spire.syntax.cfor._
 
 
-class NoDataByteGeoTiffSegment(bytes: Array[Byte], noDataValue: Byte) extends ByteGeoTiffSegment(bytes) {
-  override
-  def get(i: Int): Byte =
-    if (bytes(i) == noDataValue)
-      byteNODATA
-    else
-      bytes(i)
-}
-
-class ByteGeoTiffSegment(val bytes: Array[Byte]) extends GeoTiffSegment {
+abstract class ByteGeoTiffSegment(val bytes: Array[Byte]) extends GeoTiffSegment {
   val size: Int = bytes.size
 
-  def getInt(i: Int): Int = get(i)
-  def getDouble(i: Int): Double = i2d(get(i))
   def get(i: Int): Byte = bytes(i)
 
-  def convert(cellType: CellType): Array[Byte] =
-    cellType match {
-      case TypeBit =>
-        val bs = new BitSet(size)
-        cfor(0)(_ < size, _ + 1) { i => if ((get(i) & 1) == 0) { bs.set(i) } }
-        bs.toByteArray()
-      case TypeByte | TypeUByte =>
-        bytes
-      case TypeShort | TypeUShort =>
-        val arr = Array.ofDim[Short](size)
-        cfor(0)(_ < size, _ + 1) { i => arr(i) = i2s(get(i)) }
-        arr.toArrayByte()
-      case TypeInt =>
-        val arr = Array.ofDim[Int](size)
-        cfor(0)(_ < size, _ + 1) { i => arr(i) = getInt(i) }
-        arr.toArrayByte()
-      case TypeFloat =>
-        val arr = Array.ofDim[Float](size)
-        cfor(0)(_ < size, _ + 1) { i => arr(i) = i2f(get(i)) }
-        arr.toArrayByte()
-      case TypeDouble =>
-        val arr = Array.ofDim[Double](size)
-        cfor(0)(_ < size, _ + 1) { i => arr(i) = getDouble(i) }
-        arr.toArrayByte()
-    }
+  def getInt(i: Int): Int
+  def getDouble(i: Int): Double
+
+  protected def intToByteOut(v: Int): Byte
```

Comment:
Is this a remnant of a refactor? This is the only segment that has a method for this rather than just inlining the macros.